### PR TITLE
Split DateTimeFormat keys

### LINKED ISSUE #257: Split date time symbols into smaller data keys?
As I'm implementing Dates in DataProvider and testing them using `DateTimeFormat`, I have some questions about how should we structure that.

Generally, the data in question looks like this: https://github.com/unicode-cldr/cldr-dates-modern/tree/master/main/en

It has (per locale):
 - display names for Months, Weekdays, DayPeriods, Quarters, Eras
 - patterns for time, date, and date_time
 - list of best patterns for skeletons
 - interval patterns
 - time zone names
 - relative display names

For now, we need:
 - display names for months, weekdays, day periods
 - patterns for time, date and date_time
 - list of best patterns for skeletons

Display names come in different:
 - contexts ("format" and "stand_alone")
 - widths ("abbreviated", "narrow", "short", "wide")

but they also can be for different calendar systems (I see at least "generic" and "gregorian").

As far as I understand `DataProvider` we have some flexibility in what do we request and what we get in response.

We could, for example, put `"months/format/narrow"` as a variant in `DataEntry` and `gregory` in `DataKey` and get just a list of month names in `"format"` and `"narrow"` for `"gregory"` calendar.

Or, we can just ask for `"gregory"` and set no `variant` in `DataEntry` and get all display names for all contexts and all widths.

@sffc - what are your thoughts on that? How should a request/response look like?

### PR DESCRIPTION
**Pull request by zbraniecki**
Fixes #257

### CODE REVIEW

File: components/datetime/src/datetime.rs [hunk 1]
```diff
@@ -144,12 +164,12 @@ impl<'d> DateTimeFormat<'d> {
...
         locale: T,
         pattern: Pattern,
-        data: DataPayload<'d, 'd, DatesV1Marker>,
+        symbols_data: DataPayload<'d, 'd, DateSymbolsV1Marker>,
     ) -> Self {
         let locale = locale.into();
 
         // TODO(#257): Avoid the need for this clone by splitting the keys.
-        let symbols = DataPayload::from_owned(data.get().symbols.clone());
+        let symbols = DataPayload::from_owned(symbols_data.get().clone());
```
Comment (sffc) [inline components/datetime/src/datetime.rs:172]: Suggestion: You can delete this TODO now and just pass the `symbols_data` argument straight through. :smiley:

**Review by sffc — COMMENTED** [review]
> @sffc does it look generally like the right direction to you?

Yes, this looks like what I had in mind!

> How should I handle StructProvider in such a composed bound scenario?

I would think that in order to make a StructProvider return multiple types of structs, one would have two StructProvider instances and then tie them together with another class that delegates to one or the other.  Perhaps we could make a helper macro for that.

**Comment by zbraniecki** [comment]
This is now ready for review.

It seems to give a 5-8% win in `cargo criterion` bench for me. I'm not sure why tbh because we still load both for every `DateTimeFormat` instance, but I assume it opens up ability to later implement logic to only load symbols if needed, right?

File: components/datetime/src/datetime.rs (hunk 1 above)
Comment (sffc) [reply components/datetime/src/datetime.rs:172]: Actually, ping on this suggestion.

**Review by sffc — CHANGES_REQUESTED** [review]

**Review by sffc — APPROVED** [review]

**Review by gregtatum — APPROVED** [review]
LGTM!

**Pull request merged**